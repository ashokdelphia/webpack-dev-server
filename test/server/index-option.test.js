'use strict';

const { join } = require('path');
const request = require('supertest');
const config = require('../fixtures/index-config/webpack.config');
const testServer = require('../helpers/test-server');
const port = require('../ports-map')['index-option'];

describe('index option', () => {
  let server = null;
  let req = null;

  describe('should return index.html', () => {
    beforeAll((done) => {
      server = testServer.start(config, { port }, done);
      req = request(server.app);
    });

    it('Request to index', async () => {
      const { status, text } = await req.get('/');

      expect(status).toBe(200);
      // this file is too huge
      expect(text).not.toBe('');
    });

    afterAll(testServer.close);
  });

  describe('should return foo.html', () => {
    beforeAll((done) => {
      server = testServer.start(
        config,
        { port, index: join(__dirname, '../fixtures/index-config/foo.html') },
        done
      );
      req = request(server.app);
    });

    it('Request to index', async () => {
      const { status, text } = await req.get('/');

      expect(status).toBe(200);
      expect(text).toMatchSnapshot();
    });

    afterAll(testServer.close);
  });

  describe('should throw an error', () => {
    beforeAll((done) => {
      server = testServer.start(config, { port, index: 'foo' }, done);
      req = request(server.app);
    });

    it('Request to index', async () => {
      const { status } = await req.get('/');

      expect(status).toBe(500);
    });

    afterAll(testServer.close);
  });
});
