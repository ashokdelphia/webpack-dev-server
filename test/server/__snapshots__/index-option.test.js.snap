// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index option should return foo.html Request to index 1`] = `
"<html>
  <body>
    <p>foo</p>
  </body>
</html>
"
`;
